packages/airnode-admin/src/implementation.ts

```diff
@@ -35,6 +36,27 @@ export const parseTransactionOverrides = (args: Arguments): ethers.Overrides =>
   return overrides;
 };
 
+/**
+ * Parses Ethers transaction override options and sets maxPriorityFeePerGas to a default value if a chain supports EIP1559 transactions to prevent requesters from overpaying.
+ * @param feeData Ethers Fee Data where maxFeePerGas and maxPriorityFeePerGas will be non-null if the chain supports EIP1559 transactions.
+ * @param overrides Overrides object passed into the function
+ * @returns The parsed overrides object with values compatible with ethers
+ */
+export const parseOverrides = (feeData: ethers.providers.FeeData, overrides?: ethers.Overrides): ethers.Overrides => {
+  if (!overrides) return {};
+
+  const { gasPrice, maxFeePerGas, maxPriorityFeePerGas, ...rest } = overrides;
+  if (gasPrice && !maxFeePerGas && !maxPriorityFeePerGas && feeData.maxFeePerGas) {
+    return {
+      maxFeePerGas: gasPrice,
+      maxPriorityFeePerGas: ethers.BigNumber.from(PRIORITY_FEE),
+      ...rest,
+    };
+  }
+
+  return overrides;
+};
```

Comment:
Is this approach what was intended? Or would you prefer that we import the gas-prices implementation from `airnode-node` to set `maxFeePerGas`?

Reply:
The `if` statement scares me a bit and may be unnecessary:
The user shouldn't be specifying both eip1559 and legacy pricing override options, so I think we should error if both are specified? This is how ethers does it internally (see below).

eg.
```
if (gasPrice && (maxFeePerGas || priorityFeePerGas)) { // brackets may not be necessary
throw 'Both legacy and EIP1559 override pricing options specified - ambiguous';
}

// if the user is specifying eip1559 pricing options and the network doesn't support it,
// they shouldn't be specifying those options
if (!feeData.maxFeePerGas && (maxFeePerGas || priorityFeePerGas)) {
throw 'EIP1559 override pricing specified on legacy network'
}

/* ethers defaults to a fee of maxPriorityFeePerGas = BigNumber.from("2500000000");, 
which I think is fine for the admin utilities. The reason we went with a larger default in 
Airnode was to get in front of the herd for tx reliability, but here that's less important because
there's a human operator.
*/
return {
...feeData,
...overrides,
};
```

Ethers pricing implementation around overrides:
```
        // Do not allow mixing pre-eip-1559 and eip-1559 properties
        const hasEip1559 = (tx.maxFeePerGas != null || tx.maxPriorityFeePerGas != null);
        if (tx.gasPrice != null && (tx.type === 2 || hasEip1559)) {
            logger.throwArgumentError("eip-1559 transaction do not support gasPrice", "transaction", transaction);
        } else if ((tx.type === 0 || tx.type === 1) && hasEip1559) {
            logger.throwArgumentError("pre-eip-1559 transaction do not support maxFeePerGas/maxPriorityFeePerGas", "transaction", transaction);
        }
```

and the ethers getFeeData implementation:

```
        const { block, gasPrice } = await resolveProperties({
            block: this.getBlock("latest"),
            gasPrice: this.getGasPrice().catch((error) => {
                // @TODO: Why is this now failing on Calaveras?
                //console.log(error);
                return null;
            })
        });

        let maxFeePerGas = null, maxPriorityFeePerGas = null;

        if (block && block.baseFeePerGas) {
            // We may want to compute this more accurately in the future,
            // using the formula "check if the base fee is correct".
            // See: https://eips.ethereum.org/EIPS/eip-1559
            maxPriorityFeePerGas = BigNumber.from("2500000000");
            maxFeePerGas = block.baseFeePerGas.mul(2).add(maxPriorityFeePerGas);
        }

        return { maxFeePerGas, maxPriorityFeePerGas, gasPrice };
```